File: binary search/Python/binary_search.py
def binary_search(list, target):
  first = 0
  last = len(list) - 1

  while first <= last:
    mid_point = (first + last) // 2

    if list[mid_point] == target:
      return mid_point
    elif list[mid_point] < target:
      first = mid_point + 1
    else:
      last = mid_point - 1
  return None


def verify(index):
  "Verifies if the index exits"
  if index is not None:
    print("Target found at index ", index)
  else:
    print("Target not found.")


numbers = [num for num in range(1, 21)]

result_1 = binary_search(numbers, 5)
result_2 = binary_search(numbers,  25)

verify(result_1)
verify(result_2)

File: binary search/Java/BinarySearch.java
class BinarySearch{
  
  public static int search(int list [], int target){
    int first = 0;
    int last = list.length - 1;

    while(first <= last){
      int midPoint = (first + last) / 2;

      if(list[midPoint] == target){
        return midPoint;
      }else if(midPoint < target){
        first = midPoint + 1;
      }else{
        last = midPoint - 1;
      }
    }
    return -1;
  }


  public static void verify(int index) {
    if (index != -1) {
      System.out.println("Target found at index: " + index);
    } else {
      System.out.println("Target is not found.");
    }
  }

  public static void main(String[] args){
    int number [] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    int result_1 = search(number, 1);
    verify(result_1);

    int result_2 = search(number, 11);
    verify(result_2);
}

}